File: Backend/AstroGame/Communication/Models/Players/AddPlayerSpeciesRequest.cs
using AstroGame.Shared.Enums;
using AstroGame.Shared.Models.Players;
using AutoMapper;
using System;
using System.Collections.Generic;

namespace AstroGame.Api.Communication.Models.Players
{
    [AutoMap(typeof(PlayerSpecies))]
    public class AddPlayerSpeciesRequest
    {
        public Guid SpeciesId { get; set; }

        public string EmpireName { get; set; }

        /// <summary>
        /// The PlanetType this species prefers. 100 % habitability
        /// </summary>
        public PlanetType PreferredPlanetType { get; set; }

        public List<AddPlayerSpeciesPerkRequest> Perks { get; set; }
    }
}

File: Frontend/apps/astrogame_app/lib/communications/dtos/add_player_species_request.dart
import 'package:astrogame_app/communications/converters/guid_converter.dart';
import 'package:astrogame_app/communications/dtos/add_player_species_perk_request.dart';
import 'package:astrogame_app/models/common/guid.dart';
import 'package:astrogame_app/models/enums/planet_type.dart';

import 'package:json_annotation/json_annotation.dart';

part 'add_player_species_request.g.dart';

@GuidConverter()
@JsonSerializable()
class AddPlayerSpeciesRequest {
  Guid speciesId;
  String empireName;
  PlanetType preferredPlanetType;
  List<AddPlayerSpeciesPerkRequest> perks = [];

  AddPlayerSpeciesRequest(
      this.speciesId, this.empireName, this.preferredPlanetType, this.perks);

  factory AddPlayerSpeciesRequest.fromJson(Map<String, dynamic> json) =>
      _$AddPlayerSpeciesRequestFromJson(json);
  Map<String, dynamic> toJson() => _$AddPlayerSpeciesRequestToJson(this);
}
